docs/tutorials/losses_multi_similarity.ipynb

```diff
@@ -0,0 +1,248 @@
+    "* [Multi-Similarity Loss with General Pair Weighting\n",
+    "for Deep Metric Learning](http://openaccess.thecvf.com/content_CVPR_2019/papers/Wang_Multi-Similarity_Loss_With_General_Pair_Weighting_for_Deep_Metric_Learning_CVPR_2019_paper.pdf)\n",
+    "* [Tensorflow Implementation of Multi-Similarity Loss with General Pair Weighting for Deep Metric Learning](https://github.com/geonm/tf_ms_loss)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## MultiSimilarityLoss\n",
+    "\n",
+    "MultiSimilarityLoss is implemented in two iterative steps (i.e., mining and weighting). This allows it to fully consider three similarities for pair weighting, providing a more principled approach for collecting and weighting informative pairs. Finally, the proposed MS loss obtains new state-of-the-art performance on four image retrieval benchmarks, where it outperforms the most recent approaches.\n",
+    "![Multi Similarity Loss Steps](https://user-images.githubusercontent.com/20843596/80210333-32bde480-8651-11ea-8376-a010a113fffb.PNG)\n",
+    "\n",
+    "**The loss function is described as a Euclidean distance function:**\n",
+    "\n",
+    "![function](https://user-images.githubusercontent.com/20843596/80210696-ce4f5500-8651-11ea-87d1-cc01da64681b.PNG)\n",
+    "\n",
+    "Where $\\alpha$, $\\beta$, $\\lambda$ are hyper-parameters similar as in Binomial Deviance Loss and S<sub>ij</sub> is weight."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "  %tensorflow_version 2.x\n",
+    "except Exception:\n",
+    "  pass"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf\n",
+    "tf.__version__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorflow_addons as tfa\n",
+    "import tensorflow_datasets as tfds"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare the Data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _normalize_img(img, label):\n",
+    "    img = tf.cast(img, tf.float32) / 255.\n",
+    "    return (img, label)\n",
+    "\n",
+    "train_dataset, test_dataset = tfds.load(name=\"mnist\", split=['train', 'test'], as_supervised=True)\n",
+    "\n",
+    "# Build your input pipelines\n",
+    "train_dataset = train_dataset.shuffle(1024).batch(32)\n",
+    "train_dataset = train_dataset.map(_normalize_img)\n",
+    "\n",
+    "test_dataset = test_dataset.batch(32)\n",
+    "test_dataset = test_dataset.map(_normalize_img)"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
may you remove this block as well?

```
c = True
for v in train_dataset.as_numpy_iterator():
    if c:
        print(v[1].shape)
        c = False
```

This block seems to just be testing.

Reply:
done